pandas/tests/indexes/test_interval.py

```diff
@@ -458,7 +766,132 @@ def test_get_indexer_subintervals(self):
         expected = np.array([0, 0, 0], dtype='intp')
         tm.assert_numpy_array_equal(actual, expected)
 
+    ### NEW GET INDEXER BEHAVIOR
+
+    @pytest.mark.xfail(reason="new indexing tests for issue 16316")
+    def get_indexer_for_interval_updated_behavior(self):
+
+        index = IntervalIndex.from_tuples([(0, 2.5), (1, 3), (2, 4)], closed='right')
+
+        # single queries
+        result = index.get_indexer([Interval(1, 3, closed='right')])
+        expected = np.array([1], dtype='intp')
+        tm.assert_numpy_array_equal(result, expected)
+
+        result = index.get_indexer([Interval(1, 3, closed='left')])
```

Comment:
To make the code here a bit shorter, you can also loop over all invalid ones (up to 'multiple queries'), and then you don't need to repeat the `expected = ..` and `tm.assert_numpy_array_equal(...` lines